cmd/internal/planetscale_edge_database.go

```diff
@@ -305,11 +316,18 @@ func (p PlanetScaleEdgeDatabase) sync(ctx context.Context, tc *psdbconnect.Table
 			case binlogdata.VEventType_VGTID:
 				vgtid := event.GetVgtid().ShardGtids[0]
 				if vgtid != nil {
-					// Update cursor to new VGTID
-					tc = &psdbconnect.TableCursor{
-						Shard:    tc.Shard,
-						Keyspace: tc.Keyspace,
-						Position: vgtid.Gtid,
+					tc.Position = vgtid.Gtid
+					if vgtid.TablePKs != nil {
```

Comment:
Better to check the len() here as it could be empty and then we get a panic on the next line as there's no element at position 0.